### PR TITLE
Remove unsupported plugin from proxy in order to avoid exception catching

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Remove unsupported plugins from proxy

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/plugins/proxy/src/plugin.cpp [hunk 1]
```diff
@@ -433,10 +433,41 @@ std::string ov::proxy::Plugin::get_fallback_device(size_t idx) const {
...
+            core->get_property(device, ov::available_devices);
+        } catch (const std::runtime_error&) {
+            // Device cannot be loaded, so remove this device
+            unsupported_devices.insert(device);
+        }
+    }
+    for (const auto& device : unsupported_devices) {
+        auto it = m_alias_for.find(device);
+        if (it != m_alias_for.end()) {
+            m_alias_for.erase(it);
```
Comment (nkogteva) [inline src/plugins/proxy/src/plugin.cpp:451]: What if all of them are unavailable? I see below
```
OPENVINO_ASSERT(!m_alias_for.empty());  // alias_for cannot be empty. 1 is for fallback mode, >1 in other
```

**Review by nkogteva — COMMENTED** [review]
Is it possible to add tests?

**Review by nkogteva — APPROVED** [review]

File: src/plugins/proxy/src/plugin.cpp [hunk 2]
```diff
@@ -437,28 +447,46 @@ std::vector<std::vector<std::string>> ov::proxy::Plugin::get_hidden_devices() co
     // Proxy plugin has 2 modes of matching devices:
     //  * Fallback - in this mode we report devices only for the first hidden plugin
     //  * Alias - Case when we group all devices under one common name
-    std::vector<std::vector<std::string>> result;
+    if (m_init_devs)
+        return m_hidden_devices;
+
+    std::lock_guard<std::mutex> lock(m_init_devs_mutex);
+
+    if (m_init_devs)
```
Comment (vurusovs) [inline src/plugins/proxy/src/plugin.cpp:455]: duplicate of line 450?

File: src/plugins/proxy/tests/load_proxy_plugin_tests.cpp [hunk 3]
```diff
@@ -148,6 +148,72 @@ TEST_F(ProxyTests, get_available_devices_with_low_level_plugin) {
     EXPECT_TRUE(mock_reference_dev.empty());
 }
 
+TEST_F(ProxyTests, load_proxy_without_sevelal_devices) {
```
Comment (vurusovs) [inline src/plugins/proxy/tests/load_proxy_plugin_tests.cpp:151]:
```suggestion
TEST_F(ProxyTests, load_proxy_without_several_devices) {
```

File: src/plugins/proxy/src/plugin.cpp (hunk 2 above)
Comment (ilyachur) [reply src/plugins/proxy/src/plugin.cpp:461]: It is needed to avoid data races.

**Pull request merged**